### PR TITLE
[ReactUpdateQueue.js] Remove unreachable code

### PR DESCRIPTION
**Pull request by NE-SmallTown**
1. Because the `updateExpirationTime > renderExpirationTime` will always be `false`, so I remove the branch. If this is wrong, maybe we need add some tests. But IMO it really should always be `false` because the `update` is created by _Captured_(i.e. render throw), so the `updateExpirationTime` should always equals `renderExpirationTime` when we iterate CapturedUpdate

2. Because we won't enter the branch, the `newFirstCapturedUpdate` will always be `null`, so I just remove it.

### CODE REVIEW

File: packages/react-reconciler/src/ReactUpdateQueue.js [hunk 1]
```diff
@@ -526,20 +502,16 @@ export function processUpdateQueue<State>(
...
+  if (newFirstUpdate === null) {
     // We processed every update, without skipping. That means the new base
     // state is the same as the result state.
     newBaseState = resultState;
   }
 
   queue.baseState = newBaseState;
   queue.firstUpdate = newFirstUpdate;
-  queue.firstCapturedUpdate = newFirstCapturedUpdate;
+  queue.firstCapturedUpdate = null;
```
Comment (NE-SmallTown) [inline packages/react-reconciler/src/ReactUpdateQueue.js]: I don't know why even I remove this line i.e. `queue.firstCapturedUpdate = newFirstCapturedUpdate`, it still works, maybe the queue has been set to `null` after we executed this function? I don't know, don't dig into it. So I still remain this line

**Comment by acdlite** [comment]
I will review this PR if you write a unit test that demonstrates the bug

**Comment by NE-SmallTown** [comment]
Thanks for you feedback, I'm not sure I understand, the reason I don't add a unit test(I know the PR rules) is that this is a dead code like `if (false) { ... }`, I don't know how to test this with a unit test, so IMO I think all tests passed can show that theses changes of this PR are ok as I don't add anything or modify the logic. If you could give a direction about the unit test, I will be appreciated.

**Comment by NE-SmallTown** [comment]
Hi @acdlite , do I need provide something else?

**Pull request closed without merging**